src/main/java/com/synopsys/integration/alert/component/settings/SettingsUIConfig.java

```diff
@@ -77,16 +78,90 @@ public SettingsUIConfig() {
             ldapAuthenticationType, ldapReferral, ldapUserSearchBase, ldapUserSearchFilter, ldapUserDNPatterns, ldapUserAttributes, ldapGroupSearchBase, ldapGroupSearchFilter, ldapGroupRoleAttribute, ldapRolePrefix);
     }
 
+    private Collection<String> validateProxyHost(final FieldValueModel fieldToValidate, final FieldModel fieldModel) {
+        Collection<String> result = List.of();
+        boolean hostExists = validateFieldExists(fieldModel, SettingsDescriptor.KEY_PROXY_HOST);
+        boolean portExists = validateFieldExists(fieldModel, SettingsDescriptor.KEY_PROXY_PORT);
+        boolean userNameExists = validateFieldExists(fieldModel, SettingsDescriptor.KEY_PROXY_USERNAME);
+        boolean passwordExists = validateFieldExists(fieldModel, SettingsDescriptor.KEY_PROXY_PASSWORD);
+        boolean proxyFieldExists = portExists || passwordExists || userNameExists;
+        if (proxyFieldExists && !hostExists) {
```

Comment:
Would it be clearer to just add add the last check in the line above and then have a variable that represents the data?